storage/tianmu/core/query.cpp

```diff
@@ -1107,21 +1107,18 @@ int Query::Item2CQTerm(Item *an_arg, CQTerm &term, const TabID &tmp_table, CondT
 CondID Query::ConditionNumberFromMultipleEquality(Item_equal *conds, const TabID &tmp_table, CondType filter_type,
                                                   CondID *and_me_filter, bool is_or_subtree) {
   CQTerm zero_term, first_term, next_term;
-  // stonedb8 start
-  List_STL_Iterator<Item_field> ifield;
-  List_STL_Iterator<Item_field> li;
-  li = conds->get_fields().begin();
-  // stonedb8 end
+  Item_field *ifield{nullptr};
+  auto li = conds->get_fields().begin();
 
   Item *const_item = conds->get_const();
   if (const_item) {
     if (!Item2CQTerm(const_item, zero_term, tmp_table, filter_type)) return CondID(-1);
   } else {
-    ifield = li++;
-    if (!Item2CQTerm(&*ifield, zero_term, tmp_table, filter_type)) return CondID(-1); // stonedb8 TODO
+    ifield = &*li++;
```

Comment:
here, using `()` to make the operator combination priority clearly.  because maybe on other compiler, the combination order is different.